test/test_cuda_extension.jl

```diff
@@ -0,0 +1,128 @@
+# This file is neither included from `runtests.jl` nor run in CI.
+#
+# Run it with `julia tests/test_cuda_extension.jl`. This requires that
+# Julia is installed with juliaup and will involve downloading of a
+# lot of big artifacts. The output will contain lots of error messages
+# from caught errors; what matters is that all testsets pass.
+
+using Test
+
+juliaup_found = false
+try run(pipeline(`juliaup --version`, stdout = devnull, stderr = devnull))
+    global juliaup_found = true
+catch e
+end
```

Comment:
```suggestion
catch e
    showerror(e, Base.catch_backtrace())
end
```

Reply:
Does this provide any useful information you don't get from the
```
if !juliaup_found
    error("`juliaup` needs to be installed for the CUDA extension tests")
end
```
followup?

Anyhow, `showerror` needs an `io` first argument for this not to be a method error.

Reply:
fair enough, lets leave this as is.